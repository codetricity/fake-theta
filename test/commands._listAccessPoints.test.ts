import { request, toMatchInvalidHeaderParameterError } from './helper';

describe('POST /osc/commands/execute _listAccessPoints', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._listAccessPoints' })
      .expect(200);
    expect(res.body.results.accessPoints).toMatchSnapshot();
  });
});

describe('THETA X POST /osc/commands/execute _listAccessPoints', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._listAccessPoints' })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body.results.accessPoints).toMatchSnapshot();
  });
});

describe('THETA Z1 POST /osc/commands/execute _listAccessPoints', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._listAccessPoints' })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body.results.accessPoints).toMatchSnapshot();
  });
});

describe('Unknown Type POST /osc/commands/execute _listAccessPoints', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._listAccessPoints',
      })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
