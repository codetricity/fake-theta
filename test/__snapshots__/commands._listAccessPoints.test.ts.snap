// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute _listAccessPoints should respond with results 1`] = `
[
  {
    "connectionPriority": 3,
    "ipAddressAllocation": "dynamic",
    "security": "WPA/WPA2 PSK",
    "ssid": "Test-Access-Point3",
    "ssidStealth": false,
  },
  {
    "connectionPriority": 2,
    "defaultGateway": "192.168.1.1",
    "ipAddress": "192.168.1.254",
    "ipAddressAllocation": "static",
    "security": "WPA/WPA2 PSK",
    "ssid": "Test-Access-Point2",
    "ssidStealth": false,
    "subnetMask": "24",
  },
  {
    "connectionPriority": 1,
    "defaultGateway": "192.168.1.1",
    "ipAddress": "192.168.1.253",
    "ipAddressAllocation": "static",
    "security": "WEP",
    "ssid": "Test-Access-Point1",
    "ssidStealth": true,
    "subnetMask": "24",
  },
]
`;

exports[`THETA X POST /osc/commands/execute _listAccessPoints should respond with results 1`] = `
[
  {
    "connectionPriority": 3,
    "ipAddressAllocation": "dynamic",
    "security": "WPA/WPA2 PSK",
    "ssid": "Test-Access-Point3",
    "ssidStealth": false,
  },
  {
    "connectionPriority": 2,
    "defaultGateway": "192.168.1.1",
    "ipAddress": "192.168.1.254",
    "ipAddressAllocation": "static",
    "security": "WPA/WPA2 PSK",
    "ssid": "Test-Access-Point2",
    "ssidStealth": false,
    "subnetMask": "24",
  },
  {
    "connectionPriority": 1,
    "defaultGateway": "192.168.1.1",
    "ipAddress": "192.168.1.253",
    "ipAddressAllocation": "static",
    "security": "WEP",
    "ssid": "Test-Access-Point1",
    "ssidStealth": true,
    "subnetMask": "24",
  },
]
`;

exports[`THETA Z1 POST /osc/commands/execute _listAccessPoints should respond with results 1`] = `
[
  {
    "connectionPriority": 1,
    "defaultGateway": "192.168.1.1",
    "ipAddress": "192.168.1.254",
    "ipAddressAllocation": "static",
    "security": "WEP",
    "ssid": "Test-Access-Point1",
    "ssidStealth": true,
    "subnetMask": "255.255.255.0",
  },
  {
    "connectionPriority": 2,
    "defaultGateway": "192.168.1.1",
    "ipAddress": "192.168.1.254",
    "ipAddressAllocation": "static",
    "security": "WPA/WPA2 PSK",
    "ssid": "Test-Access-Point2",
    "ssidStealth": false,
    "subnetMask": "255.255.255.0",
  },
  {
    "connectionPriority": 3,
    "ipAddressAllocation": "dynamic",
    "security": "WPA/WPA2 PSK",
    "ssid": "Test-Access-Point3",
    "ssidStealth": false,
  },
]
`;
